tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs

```diff
@@ -0,0 +1,244 @@
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Utils;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using NATS.Client.Core;
+using NATS.Client.JetStream;
+using NATS.Client.JetStream.Models;
+namespace Aspire.Hosting.Nats.Tests;
+
+public class NatsFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyNatsResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var nats = builder.AddNats("nats");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{nats.Resource.Name}"] = await nats.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddNatsClient(nats.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var natsConnection = host.Services.GetRequiredService<INatsConnection>();
+
+            var rtt = await natsConnection.PingAsync(token);
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+        var streamName = "test-stream";
+        var subjectName = "test-subject";
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        try
+        {
+            var builder1 = CreateDistributedApplicationBuilder();
+            var nats1 = builder1.AddNats("nats")
+                .WithJetStream();
+
+            if (useVolume)
+            {
+                // Use a deterministic volume name to prevent them from exhausting the machines if deletion fails
+                volumeName = VolumeNameGenerator.CreateVolumeName(nats1, nameof(WithDataShouldPersistStateBetweenUsages));
+
+                // if the volume already exists (because of a crashing previous run), try to delete it
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+                nats1.WithDataVolume(volumeName);
+            }
+            else
+            {
+                bindMountPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                nats1.WithDataBindMount(bindMountPath);
```

Comment:
Just curious - `WithDataBindMount` is mounting to `/var/lib/nats` in the container. How is this test hitting that?